drivers/classic/src/calc_air_temperature.c

```diff
@@ -176,48 +174,56 @@ HourlyT(int     Dt,
 
 /******************************************************************************
  * @brief    This function estimates the times of minimum and maximum
- *           temperature for each day of the simulation, based on the hourly
+ *           temperature for each day of the simulation, based on the diurnal
  *           cycle of incoming solar radiation.
  *****************************************************************************/
 void
-set_max_min_hour(double *hourlyrad,
-                 int     ndays,
-                 int    *tmaxhour,
-                 int    *tminhour)
+set_max_min_sec(double *subdailyrad,
+                int     ndays,
+                int     stepsize,
+                int    *tmaxsec,
+                int    *tminsec)
 {
-    int risehour;
-    int sethour;
-    int hour;
-    int i;
+    size_t step;
+    int    risesec;
+    int    setsec;
+    int    sec;
+    int    i;
+    size_t stepsperday;
+
+    stepsperday = SEC_PER_DAY / stepsize;
 
     for (i = 0; i < ndays; i++) {
-        risehour = MISSING;
-        sethour = MISSING;
-        for (hour = 0; hour < 12; hour++) {
-            if (hourlyrad[i * HOURS_PER_DAY + hour] > 0 &&
-                (i * HOURS_PER_DAY + hour == 0 ||
-                 hourlyrad[i * HOURS_PER_DAY + hour - 1] <= 0)) {
-                risehour = hour;
+        risesec = MISSING;
+        setsec = MISSING;
+        for (step = 0, sec = 0;
+             sec < 12 * SEC_PER_HOUR;
+             step++, sec += stepsize) {
+            if (subdailyrad[i * SEC_PER_DAY + step] > 0 &&
+                (i * stepsperday + step == 0 ||
+                 subdailyrad[i * stepsperday + step - 1] <= 0)) {
+                risesec = sec;
             }
         }
-        for (hour = 12; hour < HOURS_PER_DAY; hour++) {
-            if (hourlyrad[i * HOURS_PER_DAY + hour] <= 0 &&
-                hourlyrad[i * HOURS_PER_DAY + hour - 1] >
-                0) {
-                sethour = hour;
+        for (step = step + 1, sec = 12 * SEC_PER_HOUR;
+             sec < SEC_PER_DAY;
+             step++, sec += stepsize) {
+            if (subdailyrad[i * stepsperday + sec] <= 0 &&
+                subdailyrad[i * stepsperday + sec - 1] > 0) {
```

Comment:
In previous block (l. 202-205) `step` is used in indexing the array. For some reason that is not the case here (l.208-212) where `sec` is used. Is this intentional or a typo? 

In addition, the logic for iterating is slightly different than in the previous code (which did not involve the use of `step`). It would be good to add a specific test program to make sure that `hermite` does what it is supposed to and reproduces the previous answer for hourly time steps.